src/local_scheduler/format/local_scheduler.fbs

```diff
@@ -47,6 +47,9 @@ table GetTaskRequest {
   // If true, then the local scheduler will not update the actor's task
   // counter to match the assigned checkpoint index.
   actor_checkpoint_failed: bool;
+  // The list of stateful dependencies that were fulfilled by the task that
+  // this worker just finished executing.
+  execution_returns: [string];
```

Comment:
I'm a bit confused about why this needs to go through the worker at all. If I understand correctly, the following happens:

1. A local scheduler assigns a task to a worker.
2. The actor takes the last object ID from the task spec and calls that the dummy object ID.
3. The actor passes the dummy object ID back to the local scheduler in `execution_returns`.
4. The local scheduler designates that object as locally available.

Since all of the information is in the local scheduler to begin with, couldn't we just bypass the actor entirely and have the local scheduler do the bookkeeping for itself?

Reply:
For normal tasks, that's okay, but I think we'll need something like this for checkpointing. The future implementation for checkpointing should have a way for the actor to pass to the local scheduler which execution edges were included in the resumed checkpoint.

I actually agree that what you suggested is a better way to do it, and then later add a separate IPC for the special case of checkpoint resumption, but I think that will break the current checkpointing implementation. What do you think about temporarily disallowing checkpointing as part of this PR? I was going to rewrite checkpointing in my next PR anyway but wanted to get this one merged first. Another option is to do that now and fix up the current checkpointing implementation accordingly, but I'm wary of spending the effort to do that when it will get scrapped soon anyway.